### PR TITLE
Image detail-view + edit + list pages

### PR DESCRIPTION
**Pull request by ginold**
a combination of #575 and #550 with further improvements and some other fixups.

Don't know why the commits appear in disorder, they are ordered in my git log. 

Closes https://github.com/c2corg/v6_ui/issues/586

### CODE REVIEW

**Comment by tsauerwein** [comment]
FYI https://github.com/c2corg/v6_api/pull/380 and https://github.com/c2corg/v6_api/pull/379 are merged.

https://github.com/c2corg/v6_api/pull/380 adds a `creator` attribute for the image detail service:
```
"creator": {
    "username": "...",
    "user_id": 123,
    "name": "..."
},
```

And https://github.com/c2corg/v6_api/pull/379 indexes more fields for image in ES: https://github.com/c2corg/v6_api/blob/master/c2corg_api/search/mappings/image_mapping.py#L11-L20

You will have to run the `fill_index` script again to update ES:
https://github.com/c2corg/v6_api/tree/master/c2corg_api/scripts/migration#initialize-elasticsearch

**Comment by ginold** [comment]
@tsauerwein, @asaunier this pr can start being reviewed.

File: c2corg_ui/static/js/card.js [hunk 1]
```diff
@@ -119,6 +125,15 @@ app.CardController.prototype.createURL = function() {
 
 
 /**
+ * Don't create any url on edit and add pages.
```
Comment (asaunier) [inline c2corg_ui/static/js/card.js]: I don't understand what this comment means. Do you really need it here?

File: c2corg_ui/static/partials/cards/images.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    <br>
+    <span class="quality" ng-if="doc['quality']" uib-tooltip="{{'quality'| translate}}" tooltip-placement="top">
+      <span class="glyphicon glyphicon-file"></span>
+      <span x-translate>{{doc['quality']}}</span>
+    </span>
+
+    <span class="elevation-max" ng-if="doc['elevation']" uib-tooltip="{{'elevation'| translate}}">
+      <span translate class="value-title glyphicon glyphicon-sort-by-attributes"></span>
+      <span class="value">{{doc['elevation']}}&nbsp;m</span>
```
Comment (asaunier) [inline c2corg_ui/static/partials/cards/images.html]: Are you sure one need the ``elevation`` attribute for images? 
I know it is defined in the image model (https://github.com/c2corg/v6_api/blob/master/c2corg_api/models/image.py#L41 => why?) but anyway I don't think it is a relevant information in an image card.

File: c2corg_ui/static/partials/cards/images.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="outing-date">{{ doc['date_time'] | date: 'dd/MM/yyyy'}}</div>
+    <div ng-if="doc['author']"><span class="value-title" translate>author</span>: <span class="value">{{doc['author']}}</span></div>
+
+    <span class="list-item-title">{{locale.title}}</span>
+    <span class="list-item-lang" ng-if="cardCtrl.lang != locale.lang">({{locale.lang}})</span>
+
+    <br>
+    <span class="quality" ng-if="doc['quality']" uib-tooltip="{{'quality'| translate}}" tooltip-placement="top">
+      <span class="glyphicon glyphicon-file"></span>
+      <span x-translate>{{doc['quality']}}</span>
```
Comment (asaunier) [inline c2corg_ui/static/partials/cards/images.html]: Not sure "quality" is really relevant for images either.

File: c2corg_ui/static/js/auth/authservice.js [hunk 4]
```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
...
-      if (userid === users[i].id) {
-        return true;
-      }
-    }
-    return false;
-    // by default, user has rights to every doc
+
+  if (users.length) {
+    return this.hasEditRightsOuting_(users);
+  // 'default' is set when imageType is Null in the mako template
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: This comment should be in the next block.

File: c2corg_ui/static/js/auth/authservice.js [hunk 5]
```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
...
-  if (users) {
-    for (var i = 0; i < users.length; i++) {
-      if (userid === users[i].id) {
-        return true;
-      }
-    }
-    return false;
-    // by default, user has rights to every doc
+
+  if (users.length) {
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: This condition is a bit confusing. Why would a list of users mean that we are checking the rights for an outing?

File: c2corg_ui/static/js/auth/authservice.js [hunk 6]
```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
...
+    return this.hasEditRightsOuting_(users);
+  // 'default' is set when imageType is Null in the mako template
+  } else if (imageType !== 'default') {
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: What happens if ``imageType`` is not defined?

File: c2corg_ui/static/js/auth/authservice.js [hunk 7]
```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
...
+    return this.hasEditRightsImage_(imageType);
+
   } else {
+    // by default, user has rights to every doc
     return true;
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: You don't need to have an ``else`` block for a default ``return`` statement.
You can simply write:
```
if (cond A) {
  return 'foo';
}
return 'bar';
```

File: c2corg_ui/static/js/auth/authservice.js [hunk 8]
```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
...
+ * @return {boolean}
+ * @private
+ */
+app.Authentication.prototype.hasEditRightsImage_ = function(imageType) {
+  if (imageType === 'collaborative') {
+    return true;
+    // TODO: check if user == author
+  } else {
+    return false;
+  }
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]:
```
app.Authentication.prototype.hasEditRightsImage_ = function(imageType) {
  return imageType === 'collaborative';
};
```

File: c2corg_ui/static/js/documentservice.js [hunk 9]
```diff
@@ -181,7 +181,7 @@ app.Document.prototype.isCollaborative = function(type) {
     return false;
   } else if (type === 'outings' || type === 'articles') {
     // check if user == owner or participant of the outing/article
-    return this.auth_.hasEditRights(this.document.associations['users']);
+    return this.auth_.hasEditRights(this.document.associations['users'], null);
```
Comment (asaunier) [inline c2corg_ui/static/js/documentservice.js]: Do you really need to pass a ``null`` value?
``hasEditRights`` should manage itself the case where no ``imageType`` is passed.

File: c2corg_ui/static/js/user.js [hunk 10]
```diff
@@ -108,8 +108,8 @@ app.UserController.prototype.logout = function() {
  * @return {boolean}
```
Comment (asaunier) [inline c2corg_ui/static/js/user.js]: missing ``@param`` comments

File: c2corg_ui/static/js/auth/authservice.js [hunk 11]
```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
 /**
  * Checks if the current user has rights to access/edit the document.
  * TODO: rights to personal documents only for current user.
+ * @param {Array<string>} users
+ * @param {?string} imageType
  * @return {boolean}
  * @export
  */
-app.Authentication.prototype.hasEditRights = function(users) {
+app.Authentication.prototype.hasEditRights = function(users, imageType) {
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: I don't find very elegant/convenient to pass an argument that is only relevant for a given type of document.
What about for instance passing a generic ``options`` argument where you could pass the image type when needed?

File: c2corg_ui/templates/image/view.html [hunk 12]
```diff
@@ -66,59 +66,63 @@ <h1 class="routes">
...
+      <section id="document-informations" class="collapse in">
+        ${get_image_general(image)}
+        ${get_image_camera_settings(image)}
+      </section>
+    </div>
+
+      ## DESCRIPTION
+      % if locale['summary'] or locale['description']:
+      <div class="view-details-description col-xs-12  description">
+        <h3 class="heading show-phone"><span translate>personal comments</span></h3>
```
Comment (asaunier) [inline c2corg_ui/templates/image/view.html]: Rather "description" instead of "personal comments"

File: c2corg_ui/static/js/addassociation.js [hunk 13]
```diff
@@ -83,11 +83,12 @@ app.AddAssociationController = function(appApi, appDocument) {
  */
 app.AddAssociationController.prototype.associate = function(doc) {
   var parentId, childId;
+  var type = this.parentDoctype;
```
Comment (asaunier) [inline c2corg_ui/static/js/addassociation.js]: ``type`` is a bit confusing compared to ``doc['type']``. It's better to keep "parent" somewhere. At least ``parentType``.

File: c2corg_ui/templates/helpers/view.html [hunk 14]
```diff
@@ -397,11 +397,11 @@ <h3 class="heading closed show-phone" ng-click="mainCtrl.animateHeaderIcon($even
...
 ## Float buttons
-<%def name="show_float_buttons(outingUsers, doctype, doc_id, doc_lang, other_langs, missing_langs)">\
+<%def name="show_float_buttons(associatedUsers, doctype, doc_id, doc_lang, other_langs, missing_langs, image_type = 'default')">\
   <div class="float-buttons">
-      <div ng-if="::userCtrl.hasEditRights(${dumps(outingUsers)})" class="float-button float-edit"
-           tooltip-placement="left" uib-tooltip="{{'Edit' | translate}}"
-           protected-url-btn url="${request.route_path(doctype + '_edit', id=doc_id, lang=doc_lang)}">
+      <div ng-if="'${image_type}' == 'collaborative' || userCtrl.hasEditRights(${dumps(associatedUsers)})" class="float-button float-edit"
+       tooltip-placement="left" uib-tooltip="{{'Edit' | translate}}"
+       protected-url-btn url="${request.route_url(doctype + '_edit', id=doc_id, lang=doc_lang)}">
```
Comment (asaunier) [inline c2corg_ui/templates/helpers/view.html]: Are you sure of this ``request.route_url``? It looks like a regression after @tsauerwein has changed it to ``request.route_path``.

File: c2corg_ui/templates/helpers/view.html [hunk 15]
```diff
@@ -397,11 +397,11 @@ <h3 class="heading closed show-phone" ng-click="mainCtrl.animateHeaderIcon($even
 
 
 ## Float buttons
-<%def name="show_float_buttons(outingUsers, doctype, doc_id, doc_lang, other_langs, missing_langs)">\
+<%def name="show_float_buttons(associatedUsers, doctype, doc_id, doc_lang, other_langs, missing_langs, image_type = 'default')">\
```
Comment (asaunier) [inline c2corg_ui/templates/helpers/view.html]: I feel uncomfortable with using this ``default`` image type since it actually does not exist.
I would rather make it ``collaborative`` by default. Then your test below
```
ng-if="'${image_type}' == 'collaborative' || userCtrl.hasEditRights(${dumps(associatedUsers)})"
```
would become
```
% if image_type != 'collaborative' :
    ng-if="userCtrl.hasEditRights(${dumps(associatedUsers)})"
% endif
```

File: c2corg_ui/static/js/addassociation.js (hunk 13 above)
Comment (ginold) [inline c2corg_ui/static/js/addassociation.js]: `type === 'images'` is needed, but if you ask about `var type = this.parentDoctype;` it's not, I just found it shorter to write 3 times 'type' rather than 'this.parentDoctype'.

File: c2corg_ui/templates/image/filters.html [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+from c2corg_common.attributes import default_langs, activities, image_categories, image_types
+%>
+<%namespace file="../helpers/filters.html" import="add_multiselect"/>
+
+<form app-search-filters app-search-filters-controller-name="appSearchFiltersController">
+  <div class="filters">
+    <div class="row simple-filters">
+
+      <div class="filter">
+        <label translate>date_taken</label><br>
```
Comment (asaunier) [inline c2corg_ui/templates/image/filters.html]: Rather ``date_time`` as in https://github.com/c2corg/v6_api/blob/master/c2corg_api/models/image.py#L55 so that the same string is used as in the diff page?
Or a human readable string such as ``Date``?

Comment (asaunier) [inline c2corg_ui/templates/image/filters.html]: By the way I don't think the date is a "level 1" criteria for images. Not sure if the list of advanced search criteria is available anywhere.

Here is my suggestions for level 1 criterias (always displayed): 
* title
* categories
* activities

File: c2corg_ui/templates/image/filters.html [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+    <div class="row collapse" id="moreFilters">
+
+      <div class="col-xs-12 col-sm-6 filter">
+        <label translate>title</label>
+        <input type="text" class="form-control" ng-model="filtersCtrl.filters.q">
+      </div>
+
+      <div class="col-xs-12 col-sm-6 filter">
+        <label translate>elevation</label><br>
```
Comment (asaunier) [inline c2corg_ui/templates/image/filters.html]: Searching on elevation for images is very rare. I think we can remove this filter.

File: c2corg_ui/templates/image/filters.html [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        <app-slider filter="oalt" filters-list="filtersCtrl.filters" max="8850"></app-slider>
+      </div>
+
+      <div class="col-xs-12 col-sm-6 filter">
+        <label translate>type</label><br>
+        ${add_multiselect('im_typ', image_types)}
+      </div>
+
+      <div class="filters-bottom-buttons">
```
Comment (asaunier) [inline c2corg_ui/templates/image/filters.html]: Perhaps we could create helper functions for those buttons ("show results", "clear filters", "all filters") since they will be used "as is" for all doc types? Since it would impact other documents as well, we could do it in a dedicated issue/PR.

File: c2corg_ui/templates/image/helpers/detailed_images_attributes.html [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+    <span class="glyphicon glyphicon-info-sign"></span>
+    <p translate>General</p>
+  </div>
+  <span class="detail-text accordion">
+
+    % if image.get('activities') :
+      <article class="value activities">
+        <span translate class="value-title">activities</span><br>
+        % for type in image.get('activities') :
+        <span x-translate  class="route-activity icon-${type}" uib-tooltip="{{mainCtrl.translate('${type}')}}" ></span>
```
Comment (asaunier) [inline c2corg_ui/templates/image/helpers/detailed_images_attributes.html]: Is ``x-translate`` really needed here? I see no string to be translated.

File: c2corg_ui/static/js/addassociation.js (hunk 13 above)
Comment (asaunier) [inline c2corg_ui/static/js/addassociation.js]: Please use ``parentType`` at least. Else it is difficult to understand the type of what document you are testing.

File: c2corg_ui/static/js/card.js (hunk 1 above)
Comment (ginold) [inline c2corg_ui/static/js/card.js]: Maybe better to put it just before the if statement.

File: c2corg_ui/static/js/auth/authservice.js (hunk 5 above)
Comment (ginold) [inline c2corg_ui/static/js/auth/authservice.js]: because for now only outings require the list of users to know if you can edit them or not.

File: c2corg_ui/templates/image/view.html [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+    % if not version:
+       , "associations": ${dumps(image.get('associations')) | n}
+    % endif
+  });
+</%block>
+
+<header class="view-details-title">
+  <h1 class="routes">
+    <div class="icons-header">${get_route_activities(image, 'top')}</div>
+    <span class="title">${show_route_title(locale)}</span><br>
```
Comment (asaunier) [inline c2corg_ui/templates/image/view.html]: Are you sure of this "route" helpers?
By the way shouldn't we renamed ``get_route_activities`` to ``get_activities`` since it may be used for routes but also outings or images?

File: c2corg_ui/templates/image/view.html [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+    <label class="badge" onClick="window.location='${request.route_url('images_index')}' ">
```
Comment (asaunier) [inline c2corg_ui/templates/image/view.html]: Why not using a simple HTML link here instead of an angular click handler?

File: c2corg_ui/templates/image/view.html [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+    <section id="document-informations" class="collapse in">
+      ${get_image_general(image)}
+      ${get_image_camera_settings(image)}
+    </section>
+  </div>
+
+  ## DESCRIPTION
+  % if locale['summary'] or locale['description']:
+  <div class="view-details-description col-xs-12  description">
+    <h3 class="heading show-phone"><span translate>personal comments</span></h3>
```
Comment (asaunier) [inline c2corg_ui/templates/image/view.html]: ``personal comments`` => ``description``?

File: c2corg_ui/templates/image/view.html [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+        % endif
+
+      </div>
+    </span>
+  </div>
+  % endif
+
+  <div class="view-details-description col-xs-12  description">
+    <span class="lead">
+      <img class="image" src="${image_url}/${image['filename']}" alt="${locale['title']}">
```
Comment (asaunier) [inline c2corg_ui/templates/image/view.html]: Do you use the full size file here?
If yes, it's probably better to use a resized version with a link towards the full size file. See for instance in v5: http://www.camptocamp.org/images/796074/fr/bietschhorn

File: c2corg_ui/static/js/documentediting.js [hunk 24]
```diff
@@ -151,6 +151,12 @@ app.DocumentEditingController = function($scope, $element, $attrs, appLang,
   this.submit = false;
 
   /**
+   * @type {Date}
+   * @export
+   */
+  this.today = new Date();
```
Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: Can't you directly set "today" in the template (expression) instead of here?

File: c2corg_ui/templates/image/edit.html [hunk 25]
```diff
@@ -0,0 +1,316 @@
...
+              <div class="help-block" ng-messages="editForm.iso_speed.$error">
+                <p ng-message="number" class="label label-danger" style="{{ (editForm.iso_speed.$invalid && editForm.iso_speed.$touched) ? 'display:inline' : ''}}" translate>This field must be a number.</p>
+              </div>
+            </div>
+
+            ## WIDTH
+            <div class="form-group half" ng-if="image.width">
+              <label translate>width</label>
+              <div class="input-group">
+                <input type="number" disabled min="0" minlength="1" name="width" ng-model="image.width" class="form-control" />
```
Comment (asaunier) [inline c2corg_ui/templates/image/edit.html]: This info should be automatically detected, not user-provided. Ditto for height.

File: c2corg_ui/static/js/documentediting.js [hunk 26]
```diff
@@ -504,6 +504,22 @@ app.DocumentEditingController.prototype.pushToArray = function(object, property,
   app.utils.pushToArray(object, property, value, event);
 };
 
+/**
+ * @param {number} value
+ * @export
+ */
+app.DocumentEditingController.prototype.convertExposureTime = function(value) {
+  if (value === 0) {
+    this['scope']['image']['exposure_time_converted'] = this.alerts.gettext('cannot be 0');
```
Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: Not sure to understand this line! How can an error message be saved a document attribute?

File: c2corg_ui/static/js/documentediting.js [hunk 27]
```diff
@@ -504,6 +504,22 @@ app.DocumentEditingController.prototype.pushToArray = function(object, property,
   app.utils.pushToArray(object, property, value, event);
 };
 
+/**
+ * @param {number} value
+ * @export
+ */
+app.DocumentEditingController.prototype.convertExposureTime = function(value) {
```
Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: Please consider moving this function to a dedicated ``imageediting`` controller, as already done for outings.
You could also move the ``date_time`` conversion introduced at https://github.com/c2corg/v6_ui/pull/582/commits/ff1b7f5997ba9fc2aa583d4815f43010d3f78d60

File: c2corg_ui/static/js/documentediting.js [hunk 28]
```diff
@@ -504,6 +504,22 @@ app.DocumentEditingController.prototype.pushToArray = function(object, property,
...
+  if (value === 0) {
+    this['scope']['image']['exposure_time_converted'] = this.alerts.gettext('cannot be 0');
+  } else if (value < 1 && value) {
+    this['scope']['image']['exposure_time_converted'] = '1/' + (1 / value) + ' s';
+  } else if (value >= 1) {
+    this['scope']['image']['exposure_time_converted'] = value + ' s';
+  } else if (!value) {
+    this['scope']['image']['exposure_time_converted'] = '';
```
Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: Perhaps you could factorize a bit in order to write ``this['scope']['image']['exposure_time_converted']`` only once.

File: c2corg_ui/templates/image/edit.html [hunk 29]
```diff
@@ -193,17 +195,20 @@ <h2 class="heading show-phone" translate>camera settings</h2>
...
+            <div class="form-group half" ng-class="{ 'has-error': editForm.exposure_time.$invalid, 'has-success': editForm.exposure_time.$valid && image.exposure_time}">
               <label translate>exposure_time</label>
-              <div class="input-group">
-                <span class="input-group-addon">1/</span>
-                <input type="number" min="1" name="exposure_time" ng-model="image.exposure_time" class="form-control">
-                <span class="input-group-addon">s</span>
+              <div class="input-group flex">
+                {{editCtrl.convertExposureTime(image.exposure_time)}}
+                <input type="number" min="0" max="30" name="exposure_time" ng-model="image.exposure_time" ng-change="editCtrl.convertExposureTime(image.exposure_time)" class="form-control"
+                       uib-tooltip="{{'You can type 0.003125 and it will be converted to 1/320s' | translate}}">
```
Comment (asaunier) [inline c2corg_ui/templates/image/edit.html]: This help message is a bit complicated. Do we really need it?

File: c2corg_ui/static/js/auth/authservice.js (hunk 5 above)
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: That does not look robust (and explicit for code readers).
For instance "personal articles" should also be editable by associated users only I think.

It would be clearer to check for a type for instance.

File: c2corg_ui/static/js/documentediting.js (hunk 24 above)
Comment (ginold) [inline c2corg_ui/static/js/documentediting.js]: writing for example `ng-init="today = new Date"(`) will not work.

Maybe we could use a mako variable and render it with pyton `today = ${today}` ? 
why is it more appropriate to set it in the template rather than here?

File: c2corg_ui/templates/image/edit.html (hunk 25 above)
Comment (ginold) [inline c2corg_ui/templates/image/edit.html]: they are shown only if the property already exists within the object.  If yes, it will appear in a `disabled` input, so the user cannot edit it in any case.

File: c2corg_ui/templates/image/edit.html (hunk 29 above)
Comment (ginold) [inline c2corg_ui/templates/image/edit.html]: i would have no idea that you should type the value in this format if there were no help message :P

File: c2corg_ui/static/js/documentediting.js (hunk 24 above)
Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: > Maybe we could use a mako variable and render it with pyton today = ${today} ? 

That's an option.

> why is it more appropriate to set it in the template rather than here?

Because it is needed only for images, not the other doc types.

I think you should create an ``ImageEditing`` controller. You could move this property here as well.

File: c2corg_ui/templates/image/edit.html (hunk 25 above)
Comment (asaunier) [inline c2corg_ui/templates/image/edit.html]: If you cannot edit them, why do you need to display them anyway?
It adds complexity to the editing form, and I don't think it is justified.

File: c2corg_ui/templates/image/edit.html (hunk 29 above)
Comment (asaunier) [inline c2corg_ui/templates/image/edit.html]: Perhaps, but who will type "0.003125" anyway?
Even if you know that it will be converted, no one will calculate the value of "1/320s".

File: c2corg_ui/static/js/documentediting.js (hunk 24 above)
Comment (ginold) [inline c2corg_ui/static/js/documentediting.js]: `Today` is also used for outings, but it has been moved to the `outingEditingController`. So maybe I should remove it from there and leave only `today` in `documentEditingController`?

File: c2corg_ui/templates/image/edit.html (hunk 29 above)
Comment (ginold) [inline c2corg_ui/templates/image/edit.html]: in other words this field is useless, at least if we want to stick to it within this format.

File: c2corg_ui/static/js/documentediting.js (hunk 24 above)
Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: Mmmm
I would say it's simpler to create a new controller based upon the outing editing controller and to have ``today`` in the new image editing controller as well (because this property is not needed for other document types).

File: c2corg_ui/static/js/documentediting.js (hunk 26 above)
Comment (ginold) [inline c2corg_ui/static/js/documentediting.js]: i used `aletrs.gettext` only to get a translated version of the message string t:P  but i'm not sure if that service does it like the `mainController's` `translate()`

Comment (asaunier) [inline c2corg_ui/static/js/documentediting.js]: The alert service should be used to display alerts, not to translate strings.
Anyway I don't think it the place for adding an error message, it should be done in the form itself using the usual validation & message tools, as for the other attributes.

File: c2corg_ui/templates/image/view.html (hunk 21 above)
Comment (ginold) [inline c2corg_ui/templates/image/view.html]: I don't remember the choice for it, probably that an `a` tag is a pain in the eye to style because it has a lot of defaults styles :P

File: c2corg_ui/templates/image/view.html (hunk 20 above)
Comment (ginold) [inline c2corg_ui/templates/image/view.html]: that's probably what's happening, we created the functions when routes arrived but then it turned out to be useful somewhere else than routes. I would prefer to  do it in a separate PR because it requires to change names in lots of pages and files.

Comment (asaunier) [inline c2corg_ui/templates/image/view.html]: Could you create an issue for this? Else we will forget it.

**Comment by ginold** [comment]
I updated the PR, you can re-review again.

File: c2corg_ui/static/partials/cards/images.html [hunk 30]
```diff
@@ -0,0 +1,16 @@
+<a ng-href="{{cardCtrl.createURL()}}" ng-init="doc = cardCtrl.doc; locale = cardCtrl.locale;">
+  <img ng-src="{{cardCtrl.createImg()}}" alt="{{locale['title']}}">
```
Comment (asaunier) [inline c2corg_ui/static/partials/cards/images.html]: Isn't it still the big image file here?

File: c2corg_ui/templates/helpers/view.html [hunk 31]
```diff
@@ -401,20 +401,28 @@ <h3 class="heading closed show-phone" ng-click="mainCtrl.animateHeaderIcon($even
 
 
 ## Float buttons
-<%def name="show_float_buttons(outingUsers, doctype, doc_id, doc_lang, other_langs, missing_langs)">\
+<%def name="show_float_buttons(associatedUsers, doctype, doc_id, doc_lang, other_langs, missing_langs, image = 'null')">\
```
Comment (asaunier) [inline c2corg_ui/templates/helpers/view.html]: Rather use ``None`` as default value than ``'null'`` (string).

File: c2corg_ui/static/js/auth/authservice.js [hunk 32]
```diff
@@ -89,31 +89,68 @@ app.Authentication.prototype.isAuthenticated = function() {
...
   if (roles.indexOf('moderator') > -1) {
     return true;
   }
-  // we are checking for editing rights of an outing
-  if (users) {
-    for (var i = 0; i < users.length; i++) {
-      if (userid === users[i].id) {
+  if (doctype === 'outings') {
+    return this.hasEditRightsOuting_(options['users']);
+
```
Comment (asaunier) [inline c2corg_ui/static/js/auth/authservice.js]: Unneeded blank line

File: c2corg_ui/templates/helpers/view.html [hunk 33]
```diff
@@ -401,20 +401,28 @@ <h3 class="heading closed show-phone" ng-click="mainCtrl.animateHeaderIcon($even
...
+  <%
+      if image != "null":
```
Comment (asaunier) [inline c2corg_ui/templates/helpers/view.html]: ``if image:``

File: c2corg_ui/templates/image/helpers/detailed_images_attributes.html [hunk 34]
```diff
@@ -0,0 +1,110 @@
+<%namespace file="../../helpers/view.html" import="show_areas, show_attr, get_document_up_down, get_document_min_max"/>
+<%! import math %>
+
+## GENERAL
+<%def name="get_image_general(image)">\
+<div class="name-icon-value general" ng-click="detailsCtrl.openTab($event)">
```
Comment (asaunier) [inline c2corg_ui/templates/image/helpers/detailed_images_attributes.html]: Indenting the content of the function would be nice.

File: c2corg_ui/templates/image/helpers/detailed_images_attributes.html [hunk 35]
```diff
@@ -0,0 +1,110 @@
...
+    <p><span class="value-title" translate>image_type</span>: <span class="value">${image['image_type']}</span></p>
+    % endif
+  </span>
+</div>
+</%def>
+
+## CAMERA SETTINGS
+<%def name="get_image_camera_settings(image)">\
+
+% if image.get('camera_name') or image.get('fnumber') or image.get('exposure_time') or image.get('iso_speed') or \
```
Comment (asaunier) [inline c2corg_ui/templates/image/helpers/detailed_images_attributes.html]: Indenting this "if" block would be nice.

**Comment by asaunier** [comment]
Thanks for addressing most of my previous comments. I have added some (minor) others.

File: c2corg_ui/static/partials/cards/images.html (hunk 30 above)
Comment (ginold) [inline c2corg_ui/static/partials/cards/images.html]: No, it uses a new function `app.utils.createImageUrl(filename, suffix)` so that it will download a Medium Image (MI) in this case.

Comment (asaunier) [inline c2corg_ui/static/partials/cards/images.html]: > SM was too small and distorted

I guess you mean SI.
Isn't MI a bit too big for cards? Why SI images are distorted? You mean that's because they are squared (are they?)? What's the problem exactly?

File: c2corg_ui/templates/helpers/view.html (hunk 31 above)
Comment (ginold) [inline c2corg_ui/templates/helpers/view.html]: Ahh, i was returning `Null` and it gave an error `"Null is not defined`" and i was like....DUUH it is not defined xd

File: c2corg_ui/static/partials/cards/images.html (hunk 30 above)
Comment (ginold) [inline c2corg_ui/static/partials/cards/images.html]: I'd love to show you but the images loaded only once and i cannot find them anymore :P SI are distorted because they are simply too small, you can see, even on the cards, that they are pixelated (and they are squared as well).

Comment (asaunier) [inline c2corg_ui/static/partials/cards/images.html]: Are the cards that big?! :p 

OK let's try with MI but honestly I'm afraid those files are a bit too heavy for a list page.

Comment (asaunier) [inline c2corg_ui/static/partials/cards/images.html]: OK for testing with MI, I was confused thinking you wanted to use the BI images. MI are not that bigger than SI and preserve the whole image, on the contrary of SI that are cropped squared images.

**Comment by ginold** [comment]
Updated, should be ready for merging anytime :)  have a look.

**Comment by asaunier** [comment]
OK I will take a look shortly.

I have created a separated issue concerning the fact that currently images with no geometry are not returned by the advanced search. See https://github.com/c2corg/v6_ui/issues/605

File: c2corg_ui/templates/image/edit.html [hunk 36]
```diff
@@ -0,0 +1,315 @@
+<%!
+from c2corg_common.attributes import default_langs, activities, quality_types, image_types, image_categories \
```
Comment (asaunier) [inline c2corg_ui/templates/image/edit.html]: The trailing \ is not needed.

**Comment by asaunier** [comment]
I have seen a minor problem else looks good.
Please address this remaining problem and merge, thanks!
